src/FreeBlock.cpp

```diff
@@ -0,0 +1,91 @@
+//
+// Created by mrminemeet on 18.12.23.
+//
+
+#include "FreeBlock.hpp"
+#include "Declarations.hpp"
+#include <iostream>
+
+FreeBlock::FreeBlock(int requestedSize) : Block(nullptr) {// Explicitly null here, and set further below. Otherwise, the debug info would overwrite it.
```

Comment:
may be good to know how many bytes this takes in memory and why (like Block). This should be will be bigger as Block I guess